### ISSUE TITLE
Prebuilt version not working

### ISSUE THREAD

**Issue by exceeded** [bonesoul/CoiniumServ#883]
Not able to get the prebuilt (paid) version to work.

Configuration all seems correct, redis is working as intended in command line as well as MySQL for hybrid storage.

11:48:39 [Error] [RedisProvider] [Goldcoin] Redis storage initialization failed: 127.0.0.1:6379
StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s); to create a disconnected multiplexer, disable AbortOnConnectFail.
  at StackExchange.Redis.ConnectionMultiplexer.ConnectImpl (System.Func`1[TResult] multiplexerFactory, System.IO.TextWriter log) [0x0008f] in <3c6615c26745427097f34fa88a9ec796>:0
  at StackExchange.Redis.ConnectionMultiplexer.Connect (System.String configuration, System.IO.TextWriter log) [0x0000d] in <3c6615c26745427097f34fa88a9ec796>:0
  at CoiniumServ.Persistance.Providers.Redis.RedisClient..ctor (System.String host, System.Int32 port, System.String password, System.String extraOptions) [0x0007a] in <b1ed165f730c4b95a17aeec4faee95ec>:0
  at CoiniumServ.Persistance.Providers.Redis.RedisProvider.Initialize () [0x00022] in <b1ed165f730c4b95a17aeec4faee95ec>:0

12:00:53 [Error] [HybridStorage] [DogeCoin] An exception occured while comitting share: Object reference not set to an instance of an object

The source version doesn't appear to work for anyone, so wondering if this project works at all?

**Comment by bonesoul** [comment]
It was not possible to connect to the redis server.

Are you sure your redis is working?

**Comment by exceeded** [comment]
Definitely, CLI version is working correctly... But my version is 4.0.6. Rather than 2.6. Could that cause issues?

**Comment by bonesoul** [comment]
I've never tested it with 4.x. May be?

**Comment by exceeded** [comment]
Okay, now trying with 2.6.14:

                _._
           _.-``__ ''-._
      _.-``    `.  `_.  ''-._           Redis 2.6.14 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._
 (    '      ,       .-`  | `,    )     Running in stand alone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 8525
  `-._    `-._  `-./  _.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |           http://redis.io
  `-._    `-._`-.__.-'_.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |
  `-._    `-._`-.__.-'_.-'    _.-'
      `-._    `-.__.-'    _.-'
          `-._        _.-'
              `-.__.-'

[8525] 08 Dec 14:44:11.711 # Server started, Redis version 2.6.14
[8525] 08 Dec 14:44:11.711 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
[8525] 08 Dec 14:44:11.711 * DB loaded from disk: 0.000 seconds
[8525] 08 Dec 14:44:11.711 * The server is now ready to accept connections on port 6379

Still the same issue: 

4:44:46 [Error] [RedisProvider] [Goldcoin] Redis storage initialization failed: 127.0.0.1:6379
StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s); to create a disconnected multiplexer, disable AbortOnConnectFail.
  at StackExchange.Redis.ConnectionMultiplexer.ConnectImpl (System.Func`1[TResult] multiplexerFactory, System.IO.TextWriter log) [0x0008f] in <3c6615c26745427097f34fa88a9ec796>:0
  at StackExchange.Redis.ConnectionMultiplexer.Connect (System.String configuration, System.IO.TextWriter log) [0x0000d] in <3c6615c26745427097f34fa88a9ec796>:0
  at CoiniumServ.Persistance.Providers.Redis.RedisClient..ctor (System.String host, System.Int32 port, System.String password, System.String extraOptions) [0x0007a] in <b1ed165f730c4b95a17aeec4faee95ec>:0
  at CoiniumServ.Persistance.Providers.Redis.RedisProvider.Initialize () [0x00022] in <b1ed165f730c4b95a17aeec4faee95ec>:0

**Comment by exceeded** [comment]
In the source code you explicitly declare the version as having to be 2.6.

I don't think that would ever work, even with 2.6.0?

I'm just about to test that version so will report back.

**Comment by exceeded** [comment]
`
        private void Initialize()
        {
            try
            {

                Client = new RedisClient(_config.Host, _config.Port, _config.Password, "version=2.6");

                // select the database
                Client.Select(_config.DatabaseId);

                /*
                // check the version
                var version = GetVersion();
                if (version < _requiredMinimumVersion)
                    throw new Exception(string.Format("You are using redis version {0}, minimum required version is 2.6", version));

                _logger.Information("Redis storage initialized: {0:l}:{1}, v{2:l}.", _config.Host, _config.Port, version);
                */

                _logger.Information("Redis storage initialized: {0:l}:{1}.", _config.Host, _config.Port);
            }
            catch (Exception e)
            {
                _logger.Error(e,"Redis storage initialization failed: {0:l}:{1}", _config.Host, _config.Port);
            }
        }
`

**Comment by exceeded** [comment]
I've tried nearly every version of redis now, and other software with redis. Any ideas at all? What version are you using specifically @bonesoul?

**Comment by bonesoul** [comment]
i had success with; https://chocolatey.org/packages/redis-64

**Comment by bonesoul** [comment]
and your redis may not be listening for 127.0.0.1, make sure it listens for correct interface.

**Comment by exceeded** [comment]
Yeah for sure @bonesoul, using latest compiled version seems to be an issue. You're using Windows?

**Comment by bonesoul** [comment]
Windows, Linux.

**Comment by exceeded** [comment]
I've literally tried everything... Any ideas at all @bonesoul? 

`root@pool:/home/wg/downloads# redis-cli
127.0.0.1:6379> info
# Server
redis_version:4.0.6
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:8e823067a501a370
redis_mode:standalone
os:Linux 4.4.0-87-generic x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:5.4.0
process_id:11772
run_id:6fcba9bc47f0862f9b64d65ad142c3e4fff54108
tcp_port:6379
uptime_in_seconds:658
uptime_in_days:0
hz:10
lru_clock:3060529
executable:/usr/local/bin/redis-server
config_file:/etc/redis/redis.conf

# Clients
connected_clients:1
client_longest_output_list:0
client_biggest_input_buf:0
blocked_clients:0

# Memory
used_memory:548976
used_memory_human:536.11K
used_memory_rss:4431872
used_memory_rss_human:4.23M
used_memory_peak:610984
used_memory_peak_human:596.66K
used_memory_peak_perc:89.85%
used_memory_overhead:536734
used_memory_startup:487104
used_memory_dataset:12242
used_memory_dataset_perc:19.79%
total_system_memory:8366477312
total_system_memory_human:7.79G
used_memory_lua:37888
used_memory_lua_human:37.00K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
mem_fragmentation_ratio:8.07
mem_allocator:jemalloc-4.0.3
active_defrag_running:0
lazyfree_pending_objects:0

# Persistence
loading:0
rdb_changes_since_last_save:0
rdb_bgsave_in_progress:0
rdb_last_save_time:1513009311
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0

# Stats
total_connections_received:7
total_commands_processed:25
instantaneous_ops_per_sec:0
total_net_input_bytes:910
total_net_output_bytes:18917
instantaneous_input_kbps:0.01
instantaneous_output_kbps:6.20
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
evicted_keys:0
keyspace_hits:0
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0

# Replication
role:master
connected_slaves:0
master_replid:ede09acc48188336a1592431d0e02e3241416b0f
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.18
used_cpu_user:0.12
used_cpu_sys_children:0.00
used_cpu_user_children:0.00

# Cluster
cluster_enabled:0

# Keyspace
`

I've gone back to the latest version, you can see that redis doesn't or hasn't seen the connection attempt...

`16:30:38 [Error] [RedisProvider] [Goldcoin] Redis storage initialization failed: 127.0.0.1:6379
StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s); to create a disconnected multiplexer, disable AbortOnConnectFail.
  at StackExchange.Redis.ConnectionMultiplexer.ConnectImpl (System.Func`1[TResult] multiplexerFactory, System.IO.TextWriter log) [0x0008f] in <3c6615c26745427097f34fa88a9ec796>:0
  at StackExchange.Redis.ConnectionMultiplexer.Connect (System.String configuration, System.IO.TextWriter log) [0x0000d] in <3c6615c26745427097f34fa88a9ec796>:0
  at CoiniumServ.Persistance.Providers.Redis.RedisClient..ctor (System.String host, System.Int32 port, System.String password, System.String extraOptions) [0x0007a] in <b1ed165f730c4b95a17aeec4faee95ec>:0
  at CoiniumServ.Persistance.Providers.Redis.RedisProvider.Initialize () [0x00022] in <b1ed165f730c4b95a17aeec4faee95ec>:0
`
Is there nothing in the software that could be a cause to the problem from the stack trace provided?

Can I increase logging to help diagnose the issue?

**Comment by exceeded** [comment]
Wondering if it's possible to connect to a socket file either?

**Comment by exceeded** [comment]
I've set the redis logging to DEBUG, and seeing the following connection attempts (so it is connecting):

```
20863:M 11 Dec 16:48:42.724 - Accepted 127.0.0.1:48430
20863:M 11 Dec 16:48:42.724 - Accepted 127.0.0.1:48432
20863:M 11 Dec 16:48:44.797 - 2 clients connected (0 slaves), 529096 bytes in use
20863:M 11 Dec 16:48:47.795 - Accepted 127.0.0.1:48434
20863:M 11 Dec 16:48:47.798 - Client closed connection
20863:M 11 Dec 16:48:47.799 - Accepted 127.0.0.1:48436
20863:M 11 Dec 16:48:47.799 - Client closed connection
20863:M 11 Dec 16:48:49.820 - 2 clients connected (0 slaves), 529144 bytes in use
20863:M 11 Dec 16:48:52.850 - Client closed connection
20863:M 11 Dec 16:48:52.850 - Accepted 127.0.0.1:48438
20863:M 11 Dec 16:48:52.850 - Client closed connection
20863:M 11 Dec 16:48:52.850 - Accepted 127.0.0.1:48440
20863:M 11 Dec 16:48:54.839 - 2 clients connected (0 slaves), 528072 bytes in use
20863:M 11 Dec 16:48:57.904 - Client closed connection
20863:M 11 Dec 16:48:57.904 - Client closed connection
20863:M 11 Dec 16:48:59.875 - 0 clients connected (0 slaves), 486128 bytes in use
```

**Comment by Allar** [comment]
I'm in the exact same boat exceeded. Even down to the log you posted. Other things able to use the redis server, but precompiled binary is a no-go. I've tried every configuration I can think of.

**Comment by ftx** [comment]
Same problem :/

**Comment by bonesoul** [comment]
I'll be re-checking it.

**Comment by adambant** [comment]
I am having same exact issues. Using redis 4.0.6. Apparently adding "AbortOnConnectFail": false option to any of the pool config files as a redis option doesn't get to the code (ignored).
